resfit/fit_resonator/fit_S_data.py

```diff
@@ -788,18 +825,121 @@ class VNASweep:
     linear_amps = attr.ib(type=np.ndarray)
 
     @classmethod
-    def from_csv(cls, csv):
-        """Load data from csv file."""
-        try:
-            data = np.loadtxt(csv, delimiter=',')
-        except:
-            print("User data file not found.")
-            quit()
-        freqs = data.T[0] / 10**9
-        amps = data.T[1]
-        phases = data.T[2]
-        linear_amps = 10**(amps/20)
-        return cls(freqs=freqs,amps=amps,phases=phases,linear_amps=linear_amps)
+    def from_file(cls, file):
+        if (file[-1:]=='p' and file[-4:-2]=='.s'):
+            """Load data from .snp file."""
+            try:
+                snp = open(file, 'r')
+            except:
+                print("User data file not found.")
+                quit()
+
+            """Read in header for file"""
+            line = snp.readline()
+            while line:
+                if (line.strip()[0:1]=='#'):
+                    break
+                line = snp.readline()
+            frequency_units, data_format = get_header(line)
+
+            """Read in data segment"""
+            line = snp.readline().strip()
+            while line:
+                if line[0:1]=='!':
+                    line = snp.readline().strip()
+                    continue
+                else:
+                    break
+
+            row = line.split()
+            if row == []:
+                print("Data not found in file.")
+                quit()
+
+            if data_format == "db":
```

Comment:
Do you mind breaking these out into their own functions? It would make it a bit more readable for me. 
Something like:
```python
if data_format == 'db':
    format_data_db(row)
if data_format == 'ma':
    format_data_ma(row)
```